pandas/tests/reshape/test_pivot.py

```diff
@@ -198,6 +198,17 @@ def test_pivot_with_non_observable_dropna(self, dropna):
 
         tm.assert_frame_equal(result, expected)
 
+    def test_pivot_with_interval_index(self, dropna):
+        df = pd.DataFrame(
+            {'A': pd.Categorical([pd.Interval(0, 1)] * 4),
+             'B': [1] * 4})
```

Comment:
does this fully replicate the OP issue?

Reply:
I just added a fixture to include more scenarios.